src/java/org/apache/cassandra/utils/ExpMovingAverage.java

```diff
@@ -93,6 +94,32 @@ public MovingAverage update(double val)
         return this;
     }
 
+    /**
+     * Same as {@link #update(double)} but allows to scale the decay factor for each update separately.
+     * For integer n, has the same effect as calling update(val) n times.
+     * Useful when we want to do decay based not on the number of samples, but e.g. based on time,
+     * when time that passed between updates is variable.
+     */
+    public MovingAverage update(double val, double n)
+    {
+        Preconditions.checkArgument(n > 0.0, "n must be > 0.0");
+        double current, update;
+        do
+        {
+            current = average.get();
+
+            if (!Double.isNaN(current))
+                update = current + Math.pow(alpha, n) * (val - current);
```

Comment:
This doesn't appear to agree with the javadoc. The effect of `val` should be stronger for higher `n`, not weaker.

E.g. I believe calling `update(val)` twice results in `(1-alpha)^2 * current + (1-(1-alpha)^2) * val`, which is not the same as `(1 - alpha^2) * current + alpha^2 * val`, which this translates to.